### ISSUE TITLE
Link to homepage is not shown

### ISSUE THREAD

**Issue by sammyhenningsson** [rubygems/rubygems.org#1937]
Since a few weeks ago my "homepage" link is no longer shown. It's present in the metadata file of the gem, but it's not in the links section.

![homepage](https://user-images.githubusercontent.com/13219610/54947276-32c54f80-4f3a-11e9-992e-4ae92a1458b7.png)

```
--- !ruby/object:Gem::Specification
name: shaf
version: !ruby/object:Gem::Version
  version: 1.0.0
platform: ruby
authors:
- Sammy Henningsson
autorequire: 
bindir: bin
cert_chain:
- |
  -----BEGIN CERTIFICATE-----
  MIIDXjCCAkagAwIBAgIBATANBgkqhkiG9w0BAQsFADAsMSowKAYDVQQDDCFzYW1t
  eS5oZW5uaW5nc3Nvbi9EQz1nbWFpbC9EQz1jb20wHhcNMTgwNzE2MTYzNTMzWhcN
  MjAwNzE2MTYzNTMzWjAsMSowKAYDVQQDDCFzYW1teS5oZW5uaW5nc3Nvbi9EQz1n
  bWFpbC9EQz1jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCvkgwt
  Zn8obnCelWttayB1BrHBLUK3b8gasRRtbNk4DzAL+EW6sHSVT2u5I7Wws7JQA5VB
  NaK7tgvq3CbDLVRl9NrpDCDx09To08stPxDKi6kst1nkSPAD8g0sQlW3voeQTH98
  2Z2H3XUegHhu5Z9PU9T/7V/vZUzHPiPg1tX1JUIGOPAjVGsr7SUetbL171zK4S4Y
  tvUkIoNaph+maHttvyYB/ptiZLD53WORKd4Knw3OiJsLtrxr5hhKyQ+txQdF0P8G
  +FlR+Je7B0Ek3yg6fEiJgrdcajYCMo8Oe/GRtoHhi6J3LsYA620P1BSCddZQ2XeL
  y3rzIZyMU0iaT69nAgMBAAGjgYowgYcwCQYDVR0TBAIwADALBgNVHQ8EBAMCBLAw
  HQYDVR0OBBYEFLpI49QndnmuiDhwy2XtMps7No1pMCYGA1UdEQQfMB2BG3NhbW15
  Lmhlbm5pbmdzc29uQGdtYWlsLmNvbTAmBgNVHRIEHzAdgRtzYW1teS5oZW5uaW5n
  c3NvbkBnbWFpbC5jb20wDQYJKoZIhvcNAQELBQADggEBAFJZqH6sgeiTLvMLpxaK
  K1GaYSCyZbMutf5C3tIAgkmU5UD6B8R0bw6gTM1deM5NJ60LjzqY7rlK3YKDIbTn
  iXMCe9vd4yE/jb5Zi8Wk//9n8CMG68dQpBvmcQ58/M4gTtgsx+lIgXuI5dPQMmRi
  bhWQqqWqxT9X6njjfXFk4xn3z6mfFQNPAYqRVeTHUpXBQZPt+bYXRwHPFZGWkx4l
  BnuuhYKt3CR7YIgvnsQWlTAcU1Ipdayj6UfYqUtlc6cF3CL96NOx7mgZXV8URFiX
  ZMhjYR7sRczGJx+GxGU2EaR0bjRsPVlC4ywtFxoOfRG3WaJcpWGEoAoMJX6Z0bRv
  M40=
  -----END CERTIFICATE-----
date: 2019-03-25 00:00:00.000000000 Z
dependencies:
- !ruby/object:Gem::Dependency
  name: minitest
  requirement: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '5'
    - - ">="
      - !ruby/object:Gem::Version
        version: '5.10'
  type: :development
  prerelease: false
  version_requirements: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '5'
    - - ">="
      - !ruby/object:Gem::Version
        version: '5.10'
- !ruby/object:Gem::Dependency
  name: rack-test
  requirement: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '1.0'
  type: :development
  prerelease: false
  version_requirements: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '1.0'
- !ruby/object:Gem::Dependency
  name: rake
  requirement: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '12.0'
  type: :development
  prerelease: false
  version_requirements: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '12.0'
- !ruby/object:Gem::Dependency
  name: sequel
  requirement: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '5'
  type: :development
  prerelease: false
  version_requirements: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '5'
- !ruby/object:Gem::Dependency
  name: sinatra
  requirement: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '2.0'
  type: :development
  prerelease: false
  version_requirements: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '2.0'
- !ruby/object:Gem::Dependency
  name: sqlite3
  requirement: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '1.3'
  type: :development
  prerelease: false
  version_requirements: !ruby/object:Gem::Requirement
    requirements:
    - - "~>"
      - !ruby/object:Gem::Version
        version: '1.3'
description: A framework for building hypermedia driven APIs with sinatra and sequel.
email: sammy.henningsson@gmail.com
executables:
- shaf
extensions: []
extra_rdoc_files: []
files:
- bin/shaf
- lib/shaf.rb
- lib/shaf/api_doc/comment.rb
- lib/shaf/api_doc/document.rb
- lib/shaf/app.rb
- lib/shaf/command.rb
- lib/shaf/command/base.rb
- lib/shaf/command/console.rb
- lib/shaf/command/generate.rb
- lib/shaf/command/new.rb
- lib/shaf/command/server.rb
- lib/shaf/command/templates/Gemfile.erb
- lib/shaf/command/upgrade.rb
- lib/shaf/command/version.rb
- lib/shaf/doc_model.rb
- lib/shaf/errors.rb
- lib/shaf/extensions.rb
- lib/shaf/extensions/authorize.rb
- lib/shaf/extensions/controller_hooks.rb
- lib/shaf/extensions/current_user.rb
- lib/shaf/extensions/resource_uris.rb
- lib/shaf/extensions/symbolic_routes.rb
- lib/shaf/formable.rb
- lib/shaf/formable/builder.rb
- lib/shaf/formable/field.rb
- lib/shaf/formable/form.rb
- lib/shaf/generator.rb
- lib/shaf/generator/base.rb
- lib/shaf/generator/controller.rb
- lib/shaf/generator/helper.rb
- lib/shaf/generator/migration.rb
- lib/shaf/generator/migration/add_column.rb
- lib/shaf/generator/migration/add_index.rb
- lib/shaf/generator/migration/base.rb
- lib/shaf/generator/migration/create_table.rb
- lib/shaf/generator/migration/drop_column.rb
- lib/shaf/generator/migration/empty.rb
- lib/shaf/generator/migration/rename_column.rb
- lib/shaf/generator/migration/type.rb
- lib/shaf/generator/migration/types.rb
- lib/shaf/generator/model.rb
- lib/shaf/generator/policy.rb
- lib/shaf/generator/scaffold.rb
- lib/shaf/generator/serializer.rb
- lib/shaf/generator/templates/api/controller.rb.erb
- lib/shaf/generator/templates/api/model.rb.erb
- lib/shaf/generator/templates/api/policy.rb.erb
- lib/shaf/generator/templates/api/serializer.rb.erb
- lib/shaf/generator/templates/spec/integration_spec.rb.erb
- lib/shaf/generator/templates/spec/model.rb.erb
- lib/shaf/generator/templates/spec/serializer_spec.rb.erb
- lib/shaf/helpers.rb
- lib/shaf/helpers/cache_control.rb
- lib/shaf/helpers/http_header.rb
- lib/shaf/helpers/json_html.rb
- lib/shaf/helpers/paginate.rb
- lib/shaf/helpers/payload.rb
- lib/shaf/immutable_attr.rb
- lib/shaf/middleware.rb
- lib/shaf/middleware/request_id.rb
- lib/shaf/rake.rb
- lib/shaf/rake/db.rb
- lib/shaf/rake/test.rb
- lib/shaf/registrable_factory.rb
- lib/shaf/router.rb
- lib/shaf/settings.rb
- lib/shaf/spec.rb
- lib/shaf/spec/base.rb
- lib/shaf/spec/fixture.rb
- lib/shaf/spec/fixtures.rb
- lib/shaf/spec/http_method_utils.rb
- lib/shaf/spec/integration_spec.rb
- lib/shaf/spec/let_bang.rb
- lib/shaf/spec/model.rb
- lib/shaf/spec/payload_test.rb
- lib/shaf/spec/payload_utils.rb
- lib/shaf/spec/serializer_spec.rb
- lib/shaf/spec/system_spec.rb
- lib/shaf/tasks.rb
- lib/shaf/tasks/api_doc_task.rb
- lib/shaf/tasks/db_task.rb
- lib/shaf/tasks/routes_task.rb
- lib/shaf/tasks/test_task.rb
- lib/shaf/upgrade.rb
- lib/shaf/upgrade/manifest.rb
- lib/shaf/upgrade/package.rb
- lib/shaf/upgrade/version.rb
- lib/shaf/utils.rb
- lib/shaf/version.rb
- templates/Rakefile
- templates/api/controllers/base_controller.rb
- templates/api/controllers/docs_controller.rb
- templates/api/controllers/root_controller.rb
- templates/api/policies/base_policy.rb
- templates/api/serializers/base_serializer.rb
- templates/api/serializers/error_serializer.rb
- templates/api/serializers/form_serializer.rb
- templates/api/serializers/root_serializer.rb
- templates/api/serializers/validation_error_serializer.rb
- templates/config.ru
- templates/config/bootstrap.rb
- templates/config/customize.rb
- templates/config/database.rb
- templates/config/directories.rb
- templates/config/helpers.rb
- templates/config/initializers.rb
- templates/config/initializers/db_migrations.rb
- templates/config/initializers/hal_presenter.rb
- templates/config/initializers/logging.rb
- templates/config/initializers/sequel.rb
- templates/config/paths.rb
- templates/config/settings.yml
- templates/frontend/assets/css/main.css
- templates/frontend/views/form.erb
- templates/frontend/views/layout.erb
- templates/frontend/views/payload.erb
- templates/spec/integration/root_spec.rb
- templates/spec/serializers/root_serializer_spec.rb
- templates/spec/spec_helper.rb
- upgrades/0.4.0.tar.gz
- upgrades/0.5.0.tar.gz
- upgrades/0.6.0.tar.gz
- upgrades/1.0.0.tar.gz
homepage: https://github.com/sammyhenningsson/shaf
licenses:
- MIT
metadata:
  changelog_uri: https://github.com/sammyhenningsson/shaf/blob/master/CHANGELOG.md
post_install_message: 
rdoc_options: []
require_paths:
- lib
required_ruby_version: !ruby/object:Gem::Requirement
  requirements:
  - - ">="
    - !ruby/object:Gem::Version
      version: '2.4'
required_rubygems_version: !ruby/object:Gem::Requirement
  requirements:
  - - ">="
    - !ruby/object:Gem::Version
      version: '0'
requirements: []
rubygems_version: 3.0.1
signing_key: 
specification_version: 4
summary: Sinatra Hypermedia Api Framework
test_files: []
```

**Comment by sonalkr132** [comment]
Sorry about inconvenience @sammyhenningsson
Can you please use `metadata.homepage_uri` instead? We remove the [edit linkset form](https://blog.rubygems.org/2019/03/08/and-then-there-was-one-metadata-links.html), and it meant we had to come up with an alternate way of removing/deleting outdated links.
If you have set of any of metadata URI attributes (`changelog_uri`), we would consider it as you know about **all** of [URI attributes](https://guides.rubygems.org/specification-reference/#metadata) and unset ones are not shown.
[More context](https://github.com/rubygems/rubygems.org/pull/1900)

**Comment by sammyhenningsson** [comment]
Ah, I see. I was not aware of that. I will change to `homepage_uri`.
Thanks for the quick reply!!

**Comment by rafaelfranca** [comment]
@sonalkr132 why `metadata.homepage_uri` when there is also `homepage` in the gem root spec? I think  `homepage` should be the one being used.

See https://github.com/rails/sprockets/blob/v3.7.2/sprockets.gemspec#L35

**Comment by sonalkr132** [comment]
>  I think homepage should be the one being used.

We support both `metadata.homepage_uri` (recommended) and `homepage` keys for setting the homepage link. [sprockets page](https://rubygems.org/gems/sprockets) on rubygems.org is showing incorrect homepage because it hasn't had any new release (non beta) since we fixed the bug (#1734).
It will be fixed when you push a release or let us know and we will fix it manually for you. You can see pushing the same gemspec is picking correct homepage link [here](https://staging.rubygems.org/gems/sprockets)

**Comment by rafaelfranca** [comment]
Oh. I see. Thank you!
